lib/ansible/modules/cloud/vmware/vmware_vsan_health_facts.py

```diff
@@ -0,0 +1,173 @@
+        required: false
+        default: false
+        type: bool
+
+extends_documentation_fragment:
+    - vmware.documentation
+
+author:
+    - Erwan Quelin (@equelin)
+'''
+
+EXAMPLES = '''
+- name: Gather health facts from a vSAN's cluster
+  hosts: localhost
+  gather_facts: 'no'
+  vmware_vsan_health_facts:
+    hostname: "{{ vcenter_hostname }}"
+    username: "{{ vcenter_username }}"
+    password: "{{ vcenter_password }}"
+    cluster_name: 'vSAN01'
+    fetch_from_cache: 'False'
+  register: facts
+'''
+
+RETURN = '''
+health:
+    description: vSAN cluster health facts
+    type: dict
+    sample: {
+        "_vimtype": "vim.cluster.VsanClusterHealthSummary",
+        "burnInTest": null,
+        "clusterStatus": {
+            "_vimtype": "vim.cluster.VsanClusterHealthSystemStatusResult",
+            "goalState": "installed",
+            "status": "green",
+            "trackedHostsStatus": [
+                {
+                    "_vimtype": "vim.host.VsanHostHealthSystemStatusResult",
+                    "hostname": "esxi01.example.com",
+                    "issues": [],
+                    "status": "green"
+                },
+                {
+                    "_vimtype": "vim.host.VsanHostHealthSystemStatusResult",
+                    "hostname": "esxi04.example.com",
+                    "issues": [],
+                    "status": "green"
+                },
+                {
+                    "_vimtype": "vim.host.VsanHostHealthSystemStatusResult",
+                    "hostname": "esxi02.example.com",
+                    "issues": [],
+                    "status": "green"
+                },
+                {
+                    "_vimtype": "vim.host.VsanHostHealthSystemStatusResult",
+                    "hostname": "esxi03.example.com",
+                    "issues": [],
+                    "status": "green"
+                }
+            ],
+            "untrackedHosts": []
+        }
+    }
+'''
+
+import json
+import traceback
+
+try:
+    from pyVmomi import vim, vmodl, VmomiSupport
+    HAS_PYVMOMI = True
+    HAS_PYVMOMIJSON = hasattr(VmomiSupport, 'VmomiJSONEncoder')
+except ImportError:
+    PYVMOMI_IMP_ERR = traceback.format_exc()
+    HAS_PYVMOMI = False
+    HAS_PYVMOMIJSON = False
+
+try:
+    import vsanapiutils
+    import vsanmgmtObjects
+    HAS_VSANPYTHONSDK = True
+except ImportError:
+    VSANPYTHONSDK_IMP_ERR = traceback.format_exc()
+    HAS_VSANPYTHONSDK = False
+
+from ansible.module_utils.basic import AnsibleModule, missing_required_lib
+from ansible.module_utils.vmware import connect_to_api, vmware_argument_spec, find_cluster_by_name
+
+
+def run_module():
+    argument_spec = vmware_argument_spec()
+    argument_spec.update(dict(cluster_name=dict(required=True, type='str')))
+    argument_spec.update(dict(fetch_from_cache=dict(required=False, type='bool')))
+
+    module = AnsibleModule(argument_spec=argument_spec, supports_check_mode=False)
```

Comment:
All info modules supports check mode, 

```suggestion
    module = AnsibleModule(argument_spec=argument_spec, supports_check_mode=True)
```